prutsen/prutsen.pde

```diff
@@ -52,33 +58,46 @@ void interpretInput(float potmeter, float potmeter1) {
 }
 
 void sampleSetting(float volume, float BPM) {                                    // Creating function for adjusting volume and BPM.  
-  rateControl.value.setLastValue(BPM/audioSampleBPM);                            // ?
-  volumeControl.setValue(volume);                                                // ?
+  rateControl.value.setLastValue(BPM/audioSampleBPM);                            // Starting at the original BPM level of the sample 
+  volumeControl.setValue(volume);                                                // Boost the volume of the sample in decibel (can be negative)
 }
 
 void drawVisualisation () {                                                      // Function declaration for the visualisationSetting
-  float volumeHeight = map(volume, minGain, maxGain, 450, 50);                   // Convert the Volume value and map it
-  float BPMHeight = map(BPM, 0, maxBPM, 450, 50);                                // Convert the BPM and map it
-  ellipse(100, volumeHeight, 101, 101);                                          // Draw ellipse and adjus it vertically by volumeHeight
-  rect(250, BPMHeight, 101, 101);                                                // Draw rectangle and adjus it vertically by BPMHeight
-
-  for (int i = 0; i < 9; i = i + 1) {                                            // For loop function for the left LED red column  
-    if (i > map(volume, minGain, maxGain, 8.0, -0.5)) {                          // Creating if statement tot connect volume function to the left column
-      fill(255 - 28 * i, 0, 0);
+
+  pushMatrix();
+  translate(x, y);
+  rotate(radians(r_volume));
+  fill(255);
+  rect(-75, -5, 150, 10);
+  popMatrix();
+
+  r_volume = r_volume + map(volume, minGain, maxGain, 0, max_rotation_speed);
+
+  pushMatrix();
+  translate(z, w);
+  rotate(radians(r_BPM));
+  fill(255);
+  rect(-75, -5, 150, 10);
+  popMatrix();
+
+  r_BPM = r_BPM + map(BPM, 0, maxBPM, 0, max_rotation_speed);
```

Comment:
ev kun je dit in aparte functie zetten, `draw_rotating_bars` ofzo, en dan die for loops in `draw_center_bars`

Reply:
toch maar niet gedaan? 😛